### ISSUE TITLE
Regression in 2.3 : new parsing of multiline values breaks other things

### ISSUE THREAD

**Issue by zarqman** [bkeepers/dotenv#341]
Commit aaabca8 (PR #318 - parse multiline values), which is part of dotenv v2.3.0, causes a significant regression in parsing, affecting both commented lines and lines ending in equals.

### Steps to reproduce

Consider this sample .env file (./sampleenv):

```
PORT=3000
RACK_ENV=development
# RAILS_MAX_THREADS=2
#WEB_CONCURRENCY=
CLUSTER=d1

# NS1=d1_
# NS2=
# NS3=d3_
# NS4=d4_

APP_ID=
APP_KEY=abcd
```

### Expected behavior

Gemfile:
`gem 'dotenv', github: 'bkeepers/dotenv', ref: 'b44474f'`

Before aaabca8:
```ruby
Dotenv.load('sampleenv')
# => {"PORT"=>"3000", "RACK_ENV"=>"development", "CLUSTER"=>"d1", "APP_ID"=>"", "APP_KEY"=>"abcd"}
```

### Actual behavior

Gemfile:
`gem 'dotenv', github: 'bkeepers/dotenv', ref: 'aaabca8'`

After aaabca8:
```ruby
Dotenv.load('sampleenv')
# => {"PORT"=>"3000", "RACK_ENV"=>"development", "WEB_CONCURRENCY"=>"CLUSTER=d1", "NS2"=>"", "NS4"=>"d4_", "APP_ID"=>"APP_KEY=abcd"}
```

As you can see, multiline is causing any line to eat the '#' at the beginning of the next line, causing that line to be treated as uncommented.

Additionally, any line ending in = ends up wrapping to the next line instead of returning an empty value as it did previously.

### System configuration

dotenv 2.3.0
rails 5.2.0
ruby 2.5.1

**Comment by jonmagic** [comment]
Thank you for the bug report. Looking into potential solutions now.

**Comment by jonmagic** [comment]
We just shipped version 2.4.0 which removes the `Parse multiline values` support added in 2.3.0 and fixes the parsing regressions.

**Comment by zarqman** [comment]
thanks for the super quick resolution!
